cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -101,8 +95,12 @@ public Boolean call() throws Exception {
         if (!getMonitorClient().allSystemServicesOk()) {
           return false;
         }
-        // check that the dataset service is up
-        getNamespaceClient().list();
+        // check that the dataset service is up, and also that the default namespace exists
+        try {
+          getNamespaceClient().get(Id.Namespace.DEFAULT);
```

Comment:
We have seen that all services return OK, but dataset service is not up. That causes failures in integration tests. The failures are only in the first few tests in a build, then the service comes up, then tests start passing.